n3fit/src/n3fit/model_trainer.py

```diff
@@ -523,12 +522,43 @@ def _generate_observables(
         self._reset_observables()
         log.info("Generating layers")
 
+        # We need to transpose Experimental data, stacking over replicas
+        experiment_data = {
+            "trmask": [],
+            "expdata": [],
+            "expdata_vl": [],
+            "invcovmat": [],
+            "invcovmat_vl": [],
+        }
+
+        # Loop over datasets
+        for i in range(len(self.exp_info[0])):
+            # Loop over data fields
+            for key in experiment_data.keys():
```

Comment:
Since you are using below `experiment_data.keys()` maybe it is better

```python
# Fill all experiment information for all datasets for all replicas.
for key, info in experiment_data.items()
```

Reply:
Not sure what you mean here? At this point the values of `experiment_data` are just empty lists, and they are not used?

Reply:
That you can do `for i, e in experiment_info.itemd()` and then below do `e.append`

Reply:
done in [a4026e1](https://github.com/NNPDF/nnpdf/pull/1788/commits/a4026e1cd9cd310907de07b2d4f54ec86f2a668e)